### PR TITLE
OPENEUROPA-1832: Theming for Contextual navigation paragraph type.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: oe_theme.theme [hunk 1]
```diff
@@ -807,3 +807,24 @@ function oe_theme_preprocess_responsive_image_formatter(&$variables) {
   $variables['responsive_image']['#attributes']['class'][] = 'ecl-image';
   $variables['responsive_image']['#attributes']['class'][] = 'ecl-image--fluid';
 }
+
+/**
+ * Implements hook_preprocess_paragraph().
+ */
+function oe_theme_preprocess_paragraph__oe_contextual_navigation(array &$variables): void {
```
Comment (upchuk) [inline oe_theme.theme]: Could you move this below the rest of the preprocessors that deal with the paragraph types and adjust the docblock to be consistent with the rest? Thanks!

File: oe_theme.theme [hunk 2]
```diff
@@ -127,6 +127,27 @@ function oe_theme_preprocess_paragraph__oe_accordion(array &$variables): void {
...
 
+/**
+ * Implements hook_preprocess_paragraph__oe_contextual_navigation().
+ */
+function oe_theme_preprocess_paragraph__oe_contextual_navigation(array &$variables): void {
+  // Massage data to be compliant with ECL 'context_nav' pattern data structure.
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  $variables['label'] = $paragraph->get('field_oe_title')->value;
```
Comment (brummbar) [inline oe_theme.theme]: Label is never used, as inside the template we use directly `content.field_oe_title`.

File: tests/Kernel/Paragraphs/ParagraphsTest.php [hunk 3]
```diff
@@ -465,4 +465,53 @@ public function quoteDataProvider(): array {
...
+      'field_oe_limit' => 1,
+      'field_oe_text' => 'More links',
+    ]);
+
+    $paragraph->save();
+    $html = $this->renderParagraph($paragraph);
+
+    $crawler = new Crawler($html);
+    $actual = $crawler->filter('div.ecl-context-nav > span.ecl-context-nav__label')
+      ->text();
```
Comment (brummbar) [inline tests/Kernel/Paragraphs/ParagraphsTest.php]: Let's move all of these into the same line.

File: tests/Kernel/Paragraphs/ParagraphsTest.php [hunk 4]
```diff
@@ -465,4 +465,53 @@ public function quoteDataProvider(): array {
...
+    $this->assertEquals('Contextual navigation', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list a.ecl-link')
+      ->eq(0)
```
Comment (brummbar) [inline tests/Kernel/Paragraphs/ParagraphsTest.php]: Also all of these and the text ones.

File: templates/paragraphs/paragraph--oe-contextual-navigation.html.twig [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{#
+/**
+ * @file
+ * Theme override to display the 'contextual navigation' #type paragraph.
```
Comment (brummbar) [inline templates/paragraphs/paragraph--oe-contextual-navigation.html.twig]:
```suggestion
 * Theme override to display the 'contextual navigation' type paragraph.
```

File: tests/Kernel/Paragraphs/ParagraphsTest.php [hunk 6]
```diff
@@ -465,4 +465,45 @@ public function quoteDataProvider(): array {
...
+    $this->assertEquals('Link 1', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list a.ecl-link')->eq(1)->text();
+    $this->assertEquals('Link 2', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list li.ecl-context-nav__item--over-limit a.ecl-link')->eq(0)->text();
+    $this->assertEquals('Link 2', trim($actual));
+
+    $actual = $crawler->filter('div.ecl-context-nav ul.ecl-context-nav__list li.ecl-context-nav__more button')->text();
+    $this->assertEquals('More links', trim($actual));
```
Comment (ademarco) [inline tests/Kernel/Paragraphs/ParagraphsTest.php]: We don't test that links URLs are correctly printed out and, indeed, I don't see them in my manual tests:

image

They all point to `#` while I set them to `/node` and `http://example.com`.

File: oe_theme.theme [hunk 7]
```diff
@@ -127,6 +127,26 @@ function oe_theme_preprocess_paragraph__oe_accordion(array &$variables): void {
...
+  $paragraph = $variables['paragraph'];
+
+  $variables['limit'] = $paragraph->get('field_oe_limit')->value;
+  $variables['more_label'] = $paragraph->get('field_oe_text')->value;
+
+  $variables['items'] = [];
+  foreach (Element::children($variables['content']['field_oe_links']) as $index) {
+    $variables['items'][] = [
+      'label' => $variables['content']['field_oe_links'][$index]['#title'],
+      'url' => $variables['content']['field_oe_links'][$index]['#url'],
```
Comment (ademarco) [inline oe_theme.theme]: This should be `href` and not `url`, as per `./node_modules/@ecl/generic-component-link/generic-component-link.twig`

**Pull request merged**